browser/ai_chat/upload_file_helper.cc

```diff
@@ -81,59 +81,64 @@ void UploadFileHelper::RemoveUploadedImage(uint32_t index) {
 
 void UploadFileHelper::FileSelectionCanceled() {
   if (upload_image_callback_) {
-    std::move(upload_image_callback_)
-        .Run(std::nullopt, std::nullopt, std::nullopt);
+    std::move(upload_image_callback_).Run(nullptr);
   }
 }
 
 void UploadFileHelper::OnImageRead(
     std::tuple<std::optional<std::vector<uint8_t>>,
-               std::optional<std::string>,
+               std::string,
                std::optional<int64_t>> result) {
   auto image_data = std::get<0>(result);
-  if (!image_data) {
-    std::move(upload_image_callback_)
-        .Run(std::get<0>(result), std::get<1>(result), std::get<2>(result));
+  auto filesize = std::get<2>(result);
+  if (!image_data || !filesize) {
+    std::move(upload_image_callback_).Run(nullptr);
```

Comment:
I think there should be a return after this to avoid a crash

Reply:
good call!

Reply:
actually it already has return, I believe you see no return version from an old commit

Reply:
yep old commit, my bad.